content/reference/wadm/api.md

```diff
@@ -0,0 +1,218 @@
+{
+    "name": "..",
+    "version": "..",
+    "description": "..",
+    "deployed": true,
+    "status": "compensating|ready|failed|n/a"
+}
+```
+Undeployed models will not provide meaningful data in the `status` field.
+
+### Get Model Spec
+`wadm.api.{lattice}.model.info.{name}`
+
+Retrieves the specification of the model stored with the given name. 
+
+**Request**: `none` 
+
+**Response**:
+
+JSON serialization of [OAM model/yaml](https://github.com/wasmCloud/wadm/tree/main/oam)
+
+
+
+### Store Models
+`wadm.api.{lattice}.model.put.{name}`
+
+Stores the named model. If the model and version being submitted already exist, the request will be rejected. New versions will be appended to the history according to retention policy. Note that this won't automatically deploy a model, it only affects storage.
+
+**Request**: JSON serialization of OAM model
+
+**Response**:
+```json
+{
+    "result": "error|created|newversion",
+    "message": "..."
+}
+```
+
+The response will indicate if storage failed, if a new model was created, or if a new version of a model was pushed.
+
+
+### Delete Models
+`wadm.api.{lattice}.model.del.{name}`
+
+Deletes the named model. If a version is supplied, only that version is deleted. If no version is supplied, all versions of the model are deleted. Deleted models are immediately un-deployed if applicable. If the operation does result in an (asynchronous) attempt to undeploy, it will be indicated in the `undeploy` field.
+
+**Request**: empty, or string containing version
+
+**Response**:
+```json
+{
+    "result": "deleted|error",
+    "undeploy": true,
+    "message": "..."
+}
+```
+
+### Version History
+`wadm.api.{lattice}.model.versions.{name}`
+
+Retrieves a list of all stored versions of a given model, including when that version was created. 
+
+**Request**: empty
+
+**Response**: 
+```json
+[
+    {
+        "version": "...",
+        "created": "RFC3339",
+        "deployed": false
+    }
+]
+```
+
+Each of the items in the response list describes a revision and indicates whether that revision is the deployed one.
+
+
+## Managing Deployments
+Deployments are discrete instances of autonomous control agents that monitor an application model against the real-time, discovered state of a lattice. This agent monitors lattice events, compares lattice state against the desired state of the model, and issues lattice control interface commands in order to compensate for observed gaps.
+
+### Deploy
+`wadm.api.{lattice}.model.deploy.{name}`
+
+Deploys the indicated model. This operation is idempotent, and will simply "succeed" if a request to deploy an already deployed model is received. The previous version, if deployed, does not need to be _undeployed_. Instead, the desired state of the new version is compared against the observed state of the lattice and the autonomous agents will compensate accordingly.
+
+**Request**: `string` containing version to deploy. This version must already exist in model persistence.
```

Comment:
I'd just throw this value inside of a json object to start. We will almost certainly add things down the line as parameters for deploying a model